proto/types/action.proto

```diff
@@ -60,29 +60,56 @@ message Bucket {
   google.protobuf.Timestamp createTime = 4;
   google.protobuf.Timestamp stakeStartTime = 5;
   google.protobuf.Timestamp unstakeStartTime = 6;
-  bool nonDecay = 7;
+  bool autoStake = 7;
   string owner = 8;
 }
 
 message StakeCreate {
   string candidateName = 1;
   string stakedAmount = 2;
   uint32 stakedDuration = 3;
-  bool nonDecay = 4;
+  bool autoStake = 4;
   bytes payload = 5;
 }
 
-// add, unstake, or withdraw
+enum StakeUpdateType {
+  Unstake = 0;
+  Withdraw = 1;
+}
+
+// unstake or withdraw
 message StakeUpdate {
+  StakeUpdateType type = 1;
+  uint64 bucketIndex = 2;
+  bytes payload = 3;
+}
+
+// add the amount of bucket 
+message StakeAddDeposit {
+  uint64 bucketIndex = 1;
+  string amount = 2;
+  bytes payload = 3;
+}
+
+// restake the duration and autoStake flag of bucket 
+message StakeAgain {
   uint64 bucketIndex = 1;
-  bytes payload = 2;
+  uint32 stakedDuration = 2;
+  bool autoStake = 3;
+  bytes payload = 4;
+}
+
+enum StakeMoveType {
+  ChangeCandidate = 0;
+  TransferOwnership = 1;
 }
 
-// move the bucket to vote for another candidate
+// move the bucket for another candidate or transfer the ownership of bucket to another voter  
 message StakeMove {
-  string candidateName = 1;
-  uint64 bucketIndex = 2;
-  bytes payload = 3;
+  StakeMoveType type = 1;
+  string name = 2;
+  uint64 bucketIndex = 3;
```

Comment:
make bucketIndex = 2; consistent with other definition

Reply:
yes, fixed it on latest commit